services/app-web/src/components/FilterAccordion/FilterSelect/FilterSelect.test.tsx

```diff
@@ -0,0 +1,103 @@
+        },
+        {
+            label: 'Orange',
+            value: 'Orange',
+        },
+        {
+            label: 'Banana',
+            value: 'Banana',
+        },
+        {
+            label: 'Grape',
+            value: 'Grape',
+        },
+    ]
+    it('displays filter options', async () => {
+        renderWithProviders(
+            <FilterSelect
+                name={'fruit'}
+                label={'Fruit'}
+                filterOptions={filterOptions}
+            />,
+            {
+                apolloProvider: {
+                    mocks: [
+                        fetchCurrentUserMock({
+                            statusCode: 200,
+                        }),
+                    ],
+                },
+            }
+        )
+        const combobox = await screen.findByRole('combobox')
+
+        await selectEvent.openMenu(combobox)
+
+        await waitFor(() => {
+            expect(screen.getByText('Apple')).toBeInTheDocument()
+            expect(screen.getByText('Orange')).toBeInTheDocument()
+            expect(screen.getByText('Banana')).toBeInTheDocument()
+            expect(screen.getByText('Grape')).toBeInTheDocument()
+        })
+    })
+
+    it('displays selected options', async () => {
+        renderWithProviders(
+            <FilterSelect
+                name="fruit"
+                label="Fruit"
+                filterOptions={filterOptions}
+                toggleClearFilter={false}
+            />,
+            {
+                apolloProvider: {
+                    mocks: [
+                        fetchCurrentUserMock({
+                            statusCode: 200,
+                        }),
+                    ],
+                },
+            }
+        )
+        const combobox = await screen.findByRole('combobox')
+
+        //Open select drop down and check for options
+        await selectEvent.openMenu(combobox)
+        await waitFor(() => {
+            expect(screen.getByText('Apple')).toBeInTheDocument()
+            expect(screen.getByText('Orange')).toBeInTheDocument()
+            expect(screen.getByText('Banana')).toBeInTheDocument()
+            expect(screen.getByText('Grape')).toBeInTheDocument()
+        })
+
+        //Select two options
+        await selectEvent.select(
+            screen.getByTestId('fruit-filter-options'),
+            'Apple'
+        )
+        await selectEvent.openMenu(combobox)
+        await selectEvent.select(
+            screen.getByTestId('fruit-filter-options'),
+            'Grape'
+        )
+
+        //With dropdown closed expect only selected options to be on the page
+        await waitFor(() => {
+            expect(screen.queryByText('Apple')).toBeInTheDocument()
+            expect(screen.queryByText('Grape')).toBeInTheDocument()
+            expect(screen.queryByText('Orange')).not.toBeInTheDocument()
+            expect(screen.queryByText('Banana')).not.toBeInTheDocument()
+        })
+    })
```

Comment:
todo: Since this is intended to be reuseable component I would like to see unit tests for other props doing work here
- `toggleClearFilter`
- `label`

Reply:
Added some test for `label`. As for the `toggleClearFilter` it's being tested in `FilterAccordion.test.tsx` and `HealthPlanPackage.test.tsx`, but I was having issues figuring out how to test this toggle in the stand alone component. Particularly how to update the prop without rendering and clearing out the selection, as currently re-renders would clear the selection out. But this component is mean to be only used with `FilterAccordion`.